src/i_poll_events.hpp

```diff
@@ -30,20 +30,30 @@
 #ifndef __ZMQ_I_POLL_EVENTS_HPP_INCLUDED__
 #define __ZMQ_I_POLL_EVENTS_HPP_INCLUDED__
 
+#include "fd.hpp"
+
 namespace zmq
 {
 // Virtual interface to be exposed by object that want to be notified
 // about events on file descriptors.
 
 struct i_poll_events
 {
+    typedef void * handle_t;
+
     virtual ~i_poll_events () {}
 
     // Called by I/O thread when file descriptor is ready for reading.
-    virtual void in_event () = 0;
+    virtual void in_event (handle_t handle_) = 0;
 
     // Called by I/O thread when file descriptor is ready for writing.
-    virtual void out_event () = 0;
+    virtual void out_event (handle_t handle_) = 0;
+
+    // Called by I/O thread when file descriptor has error.
```

Comment:
I don't understand why err_event and pri_event were added. If I saw it correctly, all implementations call in_event.

Reply:
For multiple reasons:
-There is 4 events defined in zmq: https://github.com/zeromq/libzmq/blob/c8a1c4542d13b6492949e7525f4fe8da266cac2b/include/zmq.h#L461
-For that: https://github.com/zeromq/libzmq/blob/master/src/ipc_connecter.cpp#L113
  Allow a more fine grain callback
-I was changing the definitions

Reply:
Ok I see. I suppose ipc_connecter_t should be changed then to make use of this finer distinction. In the end, the comment in in_event should be removed and the implementation be removed/replaced by an assertion.